lib/iris/cube.py

```diff
@@ -3131,6 +3084,10 @@ def __eq__(self, other):
         if isinstance(other, Cube):
             result = self.metadata == other.metadata
 
+            # check the cube fill-value.
+            if result:
+                result = self.fill_value == other.fill_value
```

Comment:
It's a pretty valid question ...
```python
>>> 1 == 1.0
True
>>> import numpy as np
>>> np.array(1) == np.array(1.0)
True
>>> 
```
I suspect that the answer depends on who you ask, and what they want, and if they care ... regardless, we should try to be consistent internally and I'd bet $1 that we're not when applying equality throughout iris ... so is this a general issue that should be raised?